File: 287-find-the-duplicate-number/find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        ranges::sort(nums);


        for (int left = 0; left < nums.size(); left++) {
            int right = left + 1;
            if (nums[left] == nums[right]) return nums[left];
        }

        return -1;
    }
};

File: 287-find-the-duplicate-number/find-the-duplicate-number.rs
use std::collections::BTreeSet;

impl Solution {
    pub fn find_duplicate(nums: Vec<i32>) -> i32 {
        let mut hashing = BTreeSet::new();

        for num in nums {
            if hashing.contains(&num) {
                return num;
            }
            hashing.insert(num);
        }
        -1
    }
}